lib/puppet/provider/nssdb_certificate/certutil.rb

```diff
@@ -0,0 +1,136 @@
+Puppet::Type.type(:nssdb_certificate).provide(:certutil) do
+  commands :certutil => 'certutil'
+  commands :openssl => 'openssl'
+  commands :pk12util => 'pk12util'
+
+  def create
+    add_certificate
+    add_private_key if resource[:private_key]
+  end
+
+  def destroy
+    delete_certificate
+    delete_private_key if resource[:private_key]
+  end
+
+  def exists?
+    nssdb_content
+  end
+
+  def certificate
+    nssdb_fingerprint
+  end
+
+  def certificate=(value)
+    delete_certificate unless nssdb_content.nil?
+    add_certificate
+  end
+
+  def fingerprint(file)
```

Comment:
Should it have a rescue section? Not sure what we can expect if the file contains some invalid content should be an easy test. I'd expect it to log a warning and return nil.